auth/github/client.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2024 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package github
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/bradleyfalzon/ghinstallation/v2"
+	"golang.org/x/net/http/httpproxy"
+)
+
+const (
+	AppIDKey             = "githubAppID"
+	AppInstallationIDKey = "githubAppInstallationID"
+	AppPrivateKey        = "githubAppPrivateKey"
+	ApiURLKey            = "githubApiURL"
```

Comment:
I would rename this to `githubAppBaseURL` to have the prefix consistent for all keys.